server/memcached/src/test/java/org/infinispan/server/memcached/MemcachedFunctionalTest.java

```diff
@@ -108,14 +109,14 @@ public void testAddBasic(Method m) throws InterruptedException, ExecutionExcepti
    }
 
    public void testTouchWithExpiryUnixTime(Method m) throws InterruptedException, ExecutionException, TimeoutException {
-      int future = (int) TimeUnit.MILLISECONDS.toSeconds(System.currentTimeMillis() + 1000);
+      int future = (int) TimeUnit.MILLISECONDS.toSeconds(System.currentTimeMillis() + 2000);
       OperationFuture<Boolean> f = client.set(k(m), future, v(m));
       assertTrue(f.get(timeout, TimeUnit.SECONDS));
       f = client.touch(k(m), future + 1);
       assertTrue(f.get(timeout, TimeUnit.SECONDS));
       sleepThread(1100);
       assertEquals(v(m), client.get(k(m)));
-      sleepThread(1100);
+      sleepThreadUntil(TimeUnit.SECONDS.toMillis(future) + 1000);
```

Comment:
-100 to sleep for 3 seconds in a test, we should use ControlledTimeService instead.

Reply:
Converted to use the TimeService